has/haseveryelement.go

```diff
@@ -22,13 +23,16 @@ func EveryElement(expects ...*gocrest.Matcher) *gocrest.Matcher {
 		actualValue := reflect.ValueOf(actual)
 		switch actualValue.Kind() {
 		case reflect.Array, reflect.Slice:
+
+			if actualValue.Len() != len(expects) {
```

Comment:
Not sure if we should panic here. Panicking should be reserved for when:

1. We get an error we didn't expect
2. The programmer made an error.

In this case it's arguably not the case the programmer made an error. It is the case that the actual and expected values had different lengths, which should just fail the matcher.

(An example of a programming error would be if the user had mismatched types, e.g a unsuitable actual for this matcher e.g int64)

Reply:
Fixed 👍